test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -5222,42 +5263,74 @@ def save_configuration(self, outfile, mode='a'):
         """
         conf = {}
         sconf = {MGR_OBJ_SERVER: conf}
-
+        # save resourcedef file
         rd = os.path.join(self.pbs_conf['PBS_HOME'], 'server_priv',
                           'resourcedef')
-        self._save_config_file(conf, rd)
-
+        try:
+            with open(rd, 'r') as rdef_file:
+                rdef_str = rdef_file.read()
+            b64_rdef = base64.encodestring(rdef_str)
+            conf['resourcedef_file'] = b64_rdef
+        except BaseException:
+            self.logger.info("Resourcedef file not found")
+        # save pbs.conf file
+        cfg_path = os.path.join(os.sep, 'etc', 'pbs.conf')
+        with open(cfg_path, 'r') as p:
```

Comment:
As i suggested earlier, can you try to encode the pbs.conf dict itself (server.pbs_conf) so you can decode it into a dict which can be passed to set_pbs_config when loading it back?

Reply:
changed